### PR TITLE
Add com.writerduet.writersolo

### PR DESCRIPTION
**Pull request by exjaynine**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **I sent a message to them via email**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

This is just a draft PR. Not ready yet for submission until fully tested.

### CODE REVIEW

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

File: com.writerduet.writersolo.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
+command: writersolo
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=x11
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:12]: You don't need both. If this is electron, add x11 only and make wayland opt out.

File: com.writerduet.writersolo.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.freedesktop.portal.Background
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:18]:
```suggestion
```

File: com.writerduet.writersolo.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+command: writersolo
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=x11
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:16]:
```suggestion
```

File: com.writerduet.writersolo.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+base-version: '23.08'
+command: writersolo
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=wayland
+  - --socket=x11
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --talk-name=org.kde.StatusNotifierWatcher
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:15]: Are these talk names actually required? Did you check using dbus-monitor flatpak run --log-session-bus or d-feet?

File: com.writerduet.writersolo.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  - name: writersolo
+    buildsystem: simple
+    build-commands:
+      - ar -x WriterSolo-7.0.374-amd64.deb
+      - rm -f WriterSolo-7.0.374-amd64.deb
+      - tar -xf data.tar.xz
+      - rm -f control.tar.gz data.tar.xz debian-binary
+      - cp -r opt/* usr/* $FLATPAK_DEST
+      - rm -rf opt usr
+      - install -D -m 644 -t $FLATPAK_DEST/share/metainfo $FLATPAK_ID.metainfo.xml
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:34]: This file does not exist.

File: com.writerduet.writersolo.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      - rm -rf "${FLATPAK_DEST}/share/icons/hicolor/1024x1024/"
+
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:40]: Needs a flathub.json too

File: com.writerduet.writersolo.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        url: https://media.writerduet.com/solo_app/v7/WriterSolo-7.0.374-amd64.deb
+        sha256: c84d448b4ac35914e21fe9ad03271ff961095be0d34a348d875437f9fd946802
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:42]: Add a filename here to drop the version part. Then you won't need to change `ar -x WriterSolo-7.0.374-amd64.deb` for each update.

File: com.writerduet.writersolo.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        x-checker-data:
+          type: anitya
+          project-id: 179174
+          #url-template: https://github.com/bitwarden/clients/releases/download/desktop-v$version/Bitwarden-$version-amd64.deb
+          is-main-source: true
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:47]: This does not seem correct, it does not belong to this project.

File: com.writerduet.writersolo.yaml (hunk 6 above)
Comment (bbhtt) [reply com.writerduet.writersolo.yaml:40]: https://docs.flathub.org/docs/for-app-authors/maintenance#only-build-on-x86_64

File: com.writerduet.writersolo.yaml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.secrets
+  - --talk-name=org.freedesktop.portal.Background
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --system-talk-name=org.freedesktop.login1
+  - --filesystem=xdg-download
+rename-desktop-file: bitwarden.desktop
+rename-icon: bitwarden
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:21]: wrong name

**Comment by exjaynine** [comment]
Thanks for your patience and help bbhtt, I'm a total idiot when it comes to packaging for flathub, but I'll fix it right now.

File: com.writerduet.writersolo.yaml (hunk 8 above)
Comment (exjaynine) [reply com.writerduet.writersolo.yaml:47]: You think I should remove x-checker-data entirely?

**Comment by bbhtt** [comment]
> Please confirm your submission meets all the criteria

Please complete the checklist in the description. Also specifically 

a) did you contact the developers of this?

b) what's the license here/is redistribution allowed? If not this has to be switched to extra-data

**Comment by exjaynine** [comment]
I put it under extra-data like requested. And no, I do not have any such relationship with writerduet nor do I know about their stance on redistribution. I assume it's no different than google-chrome and how you can redistribute it.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
I don't know what the hell I was doing keeping libsecret in there, when there is no accounts at all with writersolo. Sorry about that.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

File: flathub.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+  "skip-arches": ["aarch64", "arm"]
+}
```
Comment (bbhtt) [inline flathub.json:3]:
```suggestion
{
  "only-arches": ["x86_64"]
}
```

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

File: com.writerduet.writersolo.yaml (hunk 8 above)
Comment (bbhtt) [reply com.writerduet.writersolo.yaml]: You can add one if you want https://github.com/flathub/flatpak-external-data-checker, see the readme on what to use

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
It couldn't find the file despite I think with the url and source named properly. I dunno why though.

File: com.writerduet.writersolo.yaml [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+      - rm -rf "${FLATPAK_DEST}/share/icons/hicolor/1024x1024/"
+
+    sources:
+      - type: extra-data
+        only-arches:
+          - x86_64
+        url: https://media.writerduet.com/solo_app/v7/WriterSolo-7.0.374-amd64.deb
+        sha256: c84d448b4ac35914e21fe9ad03271ff961095be0d34a348d875437f9fd946802
+        size: 177550750
+        filename: WriterSolo-7.0.374-amd64.deb
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:37]:
```suggestion
        filename: WriterSolo.deb
```

File: com.writerduet.writersolo.yaml [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  - --filesystem=xdg-download
+rename-desktop-file: writersolo.desktop
+rename-icon: writersolo
+modules:
+
+  - name: writersolo
+    buildsystem: simple
+    build-commands:
+      - ar -x WriterSolo-7.0.374-amd64.deb
+      - rm -f WriterSolo-7.0.374-amd64.deb
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:23]:
```suggestion
      - ar -x WriterSolo.deb
      - rm -f WriterSolo.deb
```

File: com.writerduet.writersolo.yaml [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+      - tar -xf data.tar.xz
+      - rm -f control.tar.gz data.tar.xz debian-binary
+      - cp -r opt/* usr/* $FLATPAK_DEST
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:26]:
```suggestion
      - cp -r opt/* $FLATPAK_DEST
```

**Comment by bbhtt** [comment]
This is missing, desktop file, appdata and icons

File: com.writerduet.writersolo.yaml [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --filesystem=xdg-download
+rename-desktop-file: writersolo.desktop
+rename-icon: writersolo
+modules:
+
+  - name: writersolo
+    buildsystem: simple
+    build-commands:
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:21]: This block should be in `apply_extra` script. 

```yaml
    sources:
      - type: script
        dest-filename: apply_extra
        commands:
           - command
```

In build commands, you want this https://github.com/flathub/com.amazon.Workspaces/blob/7006428fe14a357d0146bd66968aa2c198b8eee9/com.amazon.Workspaces.yml#L132-L140

File: com.writerduet.writersolo.yaml [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+command: writersolo
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --filesystem=xdg-download
+rename-desktop-file: writersolo.desktop
+rename-icon: writersolo
```
Comment (hfiguiere) [inline com.writerduet.writersolo.yaml:16]: neither exist at this stage.

**Comment by exjaynine** [comment]
Two things I want to note: I by accident put the more recent files such as the icon under master instead of new-pr like I should have. Secondly, this is a broken package. Even under my debian machine it doesn't properly install the executable. It's literally called '${product}-${version}-${arch}-${ext}' or something like that. It runs properly, but has to be renamed after installing it. Unless upstream fixes it, it's for me way too much of a headache to troubleshoot this. Sorry. I'll keep this open if anyone else wants to tackle it.

**Comment by exjaynine** [comment]
@bbhtt 
This took a hell of a lot of trial and error on my debian machine, but I got it to install AND run. It has a few issues, yes, but it works. Now, just hope that the bot has no issues with it either.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by hfiguiere** [comment]
this is not how you use shared-modules.

**Comment by exjaynine** [comment]
Thanks for the heads up, @hfiguiere

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Review by hfiguiere — CHANGES_REQUESTED** [review]
the appstream file is missing.

**Comment by exjaynine** [comment]
I had a feeling that was the case. I'll write one up and push it.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
@hfiguiere  Did I make a typo with the xml file?

File: com.writerduet.writersolo.metadata.xml [hunk 16]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>com.writerduet.writersolo</id>
+    <launchable type="desktop-id">com.writerduet.writersolo</launchable>
```
Comment (hfiguiere) [inline com.writerduet.writersolo.metadata.xml:4]:
```suggestion
    <launchable type="desktop-id">com.writerduet.writersolo.desktop</launchable>
```

**Comment by hfiguiere** [comment]
it's not installed.

**Comment by exjaynine** [comment]
I pushed the fixes, should make things better now.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
Hang on, found a typo I completely missed previously.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

File: writersolo.sh [hunk 17]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/bash
+
+
+
+zypak-wrapper '/app/extra/opt/WriterSolo/${productName}-${version}-${arch}.${ext}'
```
Comment (hfiguiere) [inline writersolo.sh:5]: I don't see where productName version, arch and ext are defined.

Comment (exjaynine) [reply writersolo.sh:5]: That's literally what the output binary of the executable is. It's literally named that. Not even installing the deb changed it. That's why this has been frustrating to troubleshoot.

File: com.writerduet.writersolo.yaml [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
+
+  - name: writersolo
+    buildsystem: simple
+    build-commands:
+      - install -D writersolo.sh /app/bin/runwritersolo
+      - install -Dm 644 -t /app/share/applications com.writerduet.writersolo.desktop
+      - install -Dm 644 -t /app/share/icons/hicolor/0x0/apps com.writerduet.writersolo.png
```
Comment (hfiguiere) [inline com.writerduet.writersolo.yaml:55]: 0x0 isn't a valid size (or location)

Comment (exjaynine) [reply com.writerduet.writersolo.yaml:55]: Sorry about that. Fixing it right now.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
@bbhtt @hfiguiere We got ourselves here a successful build. Still will need improvements, but it's done.

File: com.writerduet.writersolo.metainfo.xml [hunk 19]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+    <id>com.writerduet.writersolo</id>
+    <launchable type="desktop-id">com.writerduet.writersolo.desktop</launchable>
+    <name>WriterSolo</name>
+    <summary>An extensive desktop app for fully writing a screenplay or convert between various screenplay formats such as final draft fdx or fountain</summary>
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LicenseRef-proprietary</project_license>
```
Comment (bbhtt) [inline com.writerduet.writersolo.metainfo.xml:8]: Add `developer_name`

File: com.writerduet.writersolo.metainfo.xml [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+	<p>You get all the same features of WriterDuet, just without collaboration or any connection to an external server</p>
+	<p>You can seamlessly convert from fdx, to fountain, or even PDF.</p>
+    </description>
+    <screenshots>
+	<screenshot type="default">
+	    <image>https://raw.githubusercontent.com/exjaynine/flathub/new-pr/writersoloscreenshot2-1.png</image>
+	</screenshot>
+    </screenshots>
+    <content_rating type="oars-1.1">
+    </content_rating>
```
Comment (bbhtt) [inline com.writerduet.writersolo.metainfo.xml:21]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: com.writerduet.writersolo.yaml [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+          stable-only: true
+  
+  - name: zypak
+    sources:
+      - type: git
+        url: https://github.com/refi64/zypak
+        commit: 55a60d110301e8dd37b631503c3524ab7baaf7aa
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\d.]+)$
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:48]:
```suggestion
```
It's in the baseapp.

File: writersolo.sh (hunk 17 above)
Comment (bbhtt) [reply writersolo.sh:5]: Add something in apply_extra no name it correctly

File: apply_extra.sh [hunk 22]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/bash
+
+
+bsdtar -xf writersolo.deb
+tar -xvf data.tar.xz
```
Comment (bbhtt) [inline apply_extra.sh:5]: This makes the output show up on terminal while installing change it to:

```
bsdtar -Oxf writersolo.deb 'data.tar.xz'|bsdtar -xf - --exclude='usr'
```

File: apply_extra.sh [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+rm *.gz
+rm debian-binary
+
+#Note, not removing the usr/ directory it comes with. At least, not right now.
+
+
+
+
+#SUID chrome-sandbox for Electron 5+
+chmod 4755 ./opt/WriterSolo/chrome-sandbox
```
Comment (bbhtt) [inline apply_extra.sh:18]:
```suggestion
```

**Comment by exjaynine** [comment]
I'm just going to test anyways, to ensure I didn't make a typo with the bsdtar program.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
@bbhtt Any further changes need to be made.? I made a note in the shelll script that runs the binary that it's called literally something very unusual. You're free to change that at a later date to anything else, but it's fine right now as is.

**Comment by exjaynine** [comment]
@bbhtt I tested it on my local machine, it ran fine, but showed the deb being extracted in stdout when installing through the terminal. Should I tweak the apply_extra script to hide it?

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
There we go, no longer shows the unwanted output to the terminal.

File: apply_extra.sh (hunk 22 above)
Comment (exjaynine) [reply apply_extra.sh]: I removed the 2nd part, but forgot at the time to remove the -v flag to not show verbose output. I changed it later, if you're reading this now but haven't scrolled down yet.

**Comment by bbhtt** [comment]
Seems fine now. Does it require sound/mic access for anything? I'm getting a bunch of 

```
ALSA lib ../../pulse/pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused

ALSA lib ../../pulse/pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused
```

Also, did you manage to contact the developers through email/forum?

> * [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Link:

**Comment by exjaynine** [comment]
> Seems fine now. Does it require sound/mic access for anything? I'm getting a bunch of
> 
> ```
> ALSA lib ../../pulse/pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused
> 
> ALSA lib ../../pulse/pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Connection refused
> ```
> 
> Also, did you manage to contact the developers through email/forum? Just letting them know that the app is being submitted will be nice.
> 
> > * [ ]  I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Link:

I'll remove the audio permissions and right now contact writersolo, they'll get back to me pretty quickly.

**Comment by exjaynine** [comment]
Send. :)
Until they get back to me, lets hold off any merging (if it was to happen soon) because I want to see how writersolo devs respond to this. Meanwhile, I'll adjust the settings to not need any audio stuff.

**Comment by bbhtt** [comment]
> I'll remove the audio permissions

It doesn't have any right now, I'm asking if it needs that permission.

**Comment by exjaynine** [comment]
Okay, I found on potential source for audio permissions. It has on the website a feature called 'ReadAloud' which is listed as a premium feature. They told me some parts are holdovers from writerduet and they'll consider removing it. Because I have no idea if it is a holdover or not, you think I should just stick to not allowing audio permissions?

**Comment by bbhtt** [comment]
Should I remove the network permissions though?

Seems like there is an account button on bottom right with update password options which might need network.

>  you think I should just stick to not allowing audio permissions?

Clicking readaloud seems to direct users to the website, so audio is fine for now.

File: com.writerduet.writersolo.yaml [hunk 24]
```diff
@@ -0,0 +1,67 @@
...
+      - dconf-service
+      - '*.pc'
+    sources:
+      - type: archive
+        url: https://download.gnome.org/sources/dconf/0.40/dconf-0.40.0.tar.xz
+        sha256: cf7f22a4c9200421d8d3325c5c1b8b93a36843650c9f95d6451e20f0bcb24533
+        x-checker-data:
+          type: gnome
+          name: dconf
+          stable-only: true
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:39]: I don't see dconf being listed as a dependency, this is electron.

**Comment by exjaynine** [comment]
> > Should I remove the network permissions though?
> 
> Seems like there is an account button on bottom right with update password options which might need network.
> 
> > you think I should just stick to not allowing audio permissions?
> 
> Clicking readaloud seems to direct users to the website, so audio is fine for now.
I directly emailed writersolo, they told me that the accounts thing was absolutely one of the holdovers from writerduet, and it was going to be passed to the rest of the writersolo team to remove that. But for stuff like update checks, I'm going to keep network anyways.

File: com.writerduet.writersolo.yaml (hunk 24 above)
Comment (exjaynine) [reply com.writerduet.writersolo.yaml:39]: I'll remove that, I at the time was just using chrome (or was it a different one) that was very similar and had dconf as a listed dependency. I should have removed that sooner.

File: com.writerduet.writersolo.yaml [hunk 25]
```diff
@@ -0,0 +1,67 @@
...
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --filesystem=xdg-download
+modules:
+  - shared-modules/libsecret/libsecret.json
+
```
[the remark this answers is not in the document]
Comment (exjaynine) [reply com.writerduet.writersolo.yaml:17]: I can't really find any info on talk-name for tray, it says in the docs I can use abstractions like libappindicator for this.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
@bbhtt Is the 'contacting upstream' required to get their direct yes/no permission or just sending them a message that this is going to happen?

**Comment by bbhtt** [comment]
Sending a message will be fine.

**Comment by exjaynine** [comment]
> Sending a message will be fine.

In that case, all checks are marked, any additional comments or concerns before we finish this?

File: com.writerduet.writersolo.yaml [hunk 26]
```diff
@@ -0,0 +1,47 @@
...
+base-version: '23.08'
+command: runwritersolo
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --filesystem=xdg-download
+  - --talk-name=org.freedesktop
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:15]: This one is wrong.

Comment (exjaynine) [reply com.writerduet.writersolo.yaml:15]: I don't even remember why I added that, I'll remove it.

Comment (bbhtt) [reply com.writerduet.writersolo.yaml:15]: libappindicator usually needs `org.kde.StatusNotifierWatcher` and/or `com.canonical.indicator.application`. You want to check on KDE/GNOME (using extension) or other DEs that support tray icons.

Comment (exjaynine) [reply com.writerduet.writersolo.yaml:15]: Which one in particular is wrong? The XCURSOR_PATH one?

Comment (bbhtt) [reply com.writerduet.writersolo.yaml:15]: `--talk-name=org.freedesktop`

Comment (exjaynine) [reply com.writerduet.writersolo.yaml:15]: I'm going to be a bit more lenient on the filesystem and just allow $HOME directory by default. I'll fix that talk as well. Thanks for bringing it to my attention.

**Comment by exjaynine** [comment]
@bbhtt Anything else you want me to take a look at? I'll then build it.

File: com.writerduet.writersolo.yaml (hunk 26 above)
Comment (exjaynine) [reply com.writerduet.writersolo.yaml]: As for the KDE and GNOME notification ones, I'll add them right now.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

File: com.writerduet.writersolo.yaml [hunk 27]
```diff
@@ -0,0 +1,51 @@
...
+command: runwritersolo
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=x11
+  - --device=dri
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --filesystem=home
+  - --talk-name=org.freedesktop.FileManager1
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline com.writerduet.writersolo.yaml:16]:
```suggestion
```

This isn't needed, the baseapp comes with libnotify 0.8.3, notifications go through portal without any permissions.

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
@bbhtt Anything further you like to add or remove before we finish this PR?

**Comment by exjaynine** [comment]
I modify comments to better reflect the current state of the project. To also better explain things if someone else who wasn't familiar with this, was reading it for the first time.

**Comment by bbhtt** [comment]
> @bbhtt Anything further you like to add or remove before we finish this PR?

no

**Comment by exjaynine** [comment]
bot, build com.writerduet.writersolo

**Comment by exjaynine** [comment]
> > @bbhtt Anything further you like to add or remove before we finish this PR?
> 
> no

Awesome! Let me just rebuild it to reflect some changes and I say we're both good here? Thanks for all the help.

**Comment by exjaynine** [comment]
@bbhtt So now what? Do you just sign off this and it gets put on flathub website on a later date?

**Comment by exjaynine** [comment]
Do you wish for me to re-request a review from you? @bbhtt

**Comment by bbhtt** [comment]
No need, please wait.

**Comment by exjaynine** [comment]
@bbhtt I would like to thank you and @hfiguiere for your assistance and patience on helping me here. I've gone from not knowing squat on how packaging for flatpak works to getting a general rough idea of how it works. Far from an expert, but I understand how it works at least.

**Pull request closed without merging**